src/workspaces/SettingsModal/SettingsModal.tsx

```diff
@@ -0,0 +1,176 @@
+import { Ajax } from 'src/libs/ajax';
+import { withErrorReporting } from 'src/libs/error';
+import Events, { extractWorkspaceDetails } from 'src/libs/events';
+import { isFeaturePreviewEnabled } from 'src/libs/feature-previews';
+import { GCP_BUCKET_LIFECYCLE_RULES } from 'src/libs/feature-previews-config';
+import { useCancellation } from 'src/libs/react-utils';
+import BucketLifecycleSettings from 'src/workspaces/SettingsModal/BucketLifecycleSettings';
+import {
+  BucketLifecycleSetting,
+  DeleteBucketLifecycleRule,
+  isBucketLifecycleSetting,
+  isDeleteBucketLifecycleRule,
+  modifyFirstBucketDeletionRule,
+  removeFirstBucketDeletionRule,
+  suggestedPrefixes,
+  WorkspaceSetting,
+} from 'src/workspaces/SettingsModal/utils';
+import { isOwner as isWorkspaceOwner, WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+interface SettingsModalProps {
+  workspace: Workspace;
+  onDismiss: () => void;
+}
+
+/**
+ * This modal is assumed to only be displayed for Google workspaces.
+ */
+const SettingsModal = (props: SettingsModalProps): ReactNode => {
+  const { namespace, name } = props.workspace.workspace;
+  const isOwner = isWorkspaceOwner(props.workspace.accessLevel);
+
+  const [lifecycleRulesEnabled, setLifecycleRulesEnabled] = useState(false);
+  const [prefixes, setPrefixes] = useState<string[]>([]);
+  const [lifecycleAge, setLifecycleAge] = useState<number | null>(null);
+
+  // Original settings from server, may contain multiple types
+  const [workspaceSettings, setWorkspaceSettings] = useState<WorkspaceSetting[] | undefined>(undefined);
+
+  const signal = useCancellation();
+
+  const getFirstBucketLifecycleSetting = (
+    settings: WorkspaceSetting[],
+    printMultipleWarning = false
+  ): BucketLifecycleSetting | undefined => {
+    const bucketLifecycleSettings: BucketLifecycleSetting[] = settings.filter((setting: WorkspaceSetting) =>
+      isBucketLifecycleSetting(setting)
+    ) as BucketLifecycleSetting[];
+    if (bucketLifecycleSettings.length > 0) {
+      if (printMultipleWarning && bucketLifecycleSettings.length > 1) {
+        // eslint-disable-next-line no-console
+        console.log('Multiple bucket lifecycle settings found, displaying only first');
+      }
+      return bucketLifecycleSettings[0];
+    }
+    return undefined;
+  };
+
+  const getDeleteLifecycleRule = (
+    bucketLifecycleSetting: BucketLifecycleSetting | undefined,
+    printMultipleWarning = false
+  ): DeleteBucketLifecycleRule | undefined => {
+    const configRules = bucketLifecycleSetting?.config.rules;
+    const deleteRules: DeleteBucketLifecycleRule[] = configRules?.filter((rule) =>
+      isDeleteBucketLifecycleRule(rule)
+    ) as DeleteBucketLifecycleRule[];
+    if (printMultipleWarning && !!deleteRules && deleteRules.length > 1) {
+      // eslint-disable-next-line no-console
+      console.log('Multiple delete bucket lifecycle rules found, displaying only first');
```

Comment:
Would it be worth displaying this to the user at all? Or do we want to keep it simple under the assumption that most users will only have a single lifecycle rule set?

Reply:
I think we want to assume the simple case (r@admull). I just added a console log message in case someone was trying to debug an exceptional case where a person had set up multiple rules through the Swagger API.